static/components.ts

```diff
@@ -0,0 +1,866 @@
+// Copyright (c) 2021, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import {
+    EmptyCompilerState,
+    ComponentConfig,
+    PopulatedCompilerState,
+    CompilerForTreeState,
+    EmptyExecutorState,
+    PopulatedExecutorState,
+    ExecutorForTreeState,
+    EmptyEditorState,
+    PopulatedEditorState,
+    EmptyTreeState,
+    OutputState,
+    ToolViewState,
+    EmptyToolInputViewState,
+    PopulatedToolInputViewState,
+    PopulatedOptViewState,
+    EmptyOptViewState,
+    EmptyFlagsViewState,
+    PopulatedFlagsViewState,
+    EmptyDiffViewState,
+    PopulatedDiffViewState,
+    EmptyPpViewState,
+    PopulatedPpViewState,
+    EmptyAstViewState,
+    PopulatedAstViewState,
+    EmptyGccDumpViewState,
+    PopulatedGccDumpViewState,
+    GccDumpOptions,
+    EmptyCfgViewState,
+    PopulatedCfgViewState,
+    PopulatedConformanceViewState,
+    EmptyIrViewState,
+    PopulatedIrViewState,
+    EmptyRustMirViewState,
+    PopulatedRustMirViewState,
+    EmptyHaskellCoreViewState,
+    PopulatedHaskellCoreViewState,
+    EmptyHaskellStgViewState,
+    PopulatedHaskellStgViewState,
+    EmptyHaskellCmmViewState,
+    PopulatedHaskellCmmViewState,
+    EmptyGnatDebugTreeViewState,
+    PopulatedGnatDebugTreeViewState,
+    EmptyGnatDebugViewState,
+    PopulatedGnatDebugViewState,
+    EmptyRustMacroExpViewState,
+    PopulatedRustMacroExpViewState,
+    EmptyRustHirViewState,
+    PopulatedRustHirViewState,
+    EmptyDeviceViewState,
+    PopulatedDeviceViewState,
+} from './components.interfaces';
+
+export const COMPILER_COMPONENT_NAME = 'compiler';
```

Comment:
I defer to @matsjla et al; I'm still learning here - but why these exported `CONSTANT_NAMES` here and not in the interfaces?

Reply:
Good catch, I'll move them! Thanks